### ISSUE TITLE
not compatible with UglifyJs?

### ISSUE THREAD

**Issue by hontas** [developit/unfetch#93]
After adding this to my app I get the following error when building for production:

```
ERROR in js/main.js from UglifyJs
Invalid assignment [js/main.js:11866,39]
```

Using `webpack 4.19.1` and `uglifyjs-webpack-plugin 2.0.1`

Any ideas what this could be?

**Comment by developit** [comment]
How did you add Unfetch, and what version?  It's ES3, so it shouldn't have any issues with Uglify.

**Comment by hontas** [comment]
Imported like so: `import ’unfetch/polyfill’;` at the top of my entry-file. Installed it right before I reported the issue (replacing `whatwg-fetch`) so it should’ve been the latest I guess.

I’m using UglifyJs plugin in `optimization` because I also use CSS-minification - so I dont think it should run twice.

**Comment by vvxvv** [comment]
Same error from UglifyJs@3.4.9 

1. unfetch 4.0.1, import 'unfetch/polyfill'
    error: Invalid assignment  from UglifyJs

2. unfetch 3.1.2, import 'unfetch/polyfill'
    no error

3. unfetch 4.0.1, import 'unfetch/polyfill/index.js'
    no error

4. import 'whatwg-fetch';
    no error

-- 
Webpack@4.20.2, uglifyjs-webpack-plugin@2.0.1

**Comment by developit** [comment]
Looks like this may be related to `.mjs`, which shipped in Unfetch@4.

**Comment by markwainwright** [comment]
I encountered this too. Looks like it happens because:

1. Webpack resolves `unfetch/polyfill` to `unfetch/polyfill/polyfill.mjs`.
1. This file requires `src/index.js` which contains an ES6 arrow function
1. UglifyJs throws an error because it's expecting ES5.

A workaround is to change `import 'unfetch/polyfill'` to `import 'unfetch/polyfill/index'`.

@developit I'm not exactly sure what the purpose of `polyfill/polyfill.mjs` is but it seems to me that it should probably be requiring the compiled `polyfill/index.js` rather than the raw `src/index.mjs`?

**Comment by developit** [comment]
The fix for this was released in [4.2.0](https://github.com/developit/unfetch/releases/tag/4.2.0).
